lib/puppetserver/ca/list_action.rb

```diff
@@ -0,0 +1,139 @@
+require 'puppetserver/ca/utils'
+require 'puppetserver/utils/http_utilities'
+require 'puppetserver/utils/file_utilities'
+require 'puppetserver/ca/puppet_config'
+require 'optparse'
+require 'openssl'
+require 'net/https'
+require 'json'
+
+module Puppetserver
+  module Ca
+    class ListAction
+
+      include Puppetserver::Utils
+
+      SUMMARY = 'List all certificates'
+      BANNER = <<-BANNER
+Usage:
+  puppetserver ca list [--help|--version]
+  puppetserver ca list [--config]
+  puppetserver ca list [--all]
+
+Description:
+List outstanding certificates. If --all is specified, signed certificates will be listed as well.
+
+Options:
+      BANNER
+
+      BODY = JSON.dump({desired_state: 'signed'})
+
+      def initialize(logger)
+        @logger = logger
+      end
+
+      def self.parser(parsed = {})
+        OptionParser.new do |opts|
+          opts.banner = BANNER
+          opts.on('--config PUPPET.CONF', 'Custom path to Puppet\'s config file') do |conf|
+            parsed['config'] = conf
+          end
+          opts.on('--help', 'Display this command specific help output') do |help|
+            parsed['help'] = true
+          end
+          opts.on('--version', 'Output the version') do |v|
+            parsed['version'] = true
+          end
+          opts.on('--all', 'List signed certificates') do |a|
+            parsed['all'] = true
+          end
+        end
+      end
+
+      def run(input)
+        config = input['config']
+
+        if config
+          errors = FileUtilities.validate_file_paths(config)
+          return 1 if Utils.handle_errors(@logger, errors)
+        end
+
+        puppet = PuppetConfig.parse(config)
+        return 1 if Utils.handle_errors(@logger, puppet.errors)
+
+        all_certs = get_all_certs(puppet.settings)
+        requested, signed = separate_certs(all_certs)
+        if input['all']
+          human_readable(requested, signed)
+        else
+          human_readable(requested)
+        end
```

Comment:
would ``input['all'] ? human_readable(requested, signed) : human_readable(requested)`` be an improvement on this if/else block or should I leave it as is? Not sure if consolidating this would make it confusing.

Reply:
I'm fine with either